core/src/main/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformer.scala

```diff
@@ -17,72 +17,122 @@
 package io.projectglow.transformers.normalizevariants
 
 import io.projectglow.DataFrameTransformer
+import io.projectglow.common.VariantSchemas._
 import io.projectglow.common.logging.{HlsEventRecorder, HlsTagValues}
+import io.projectglow.functions.{expand_struct, normalize_variant}
+import io.projectglow.transformers.normalizevariants.VariantNormalizer._
+import io.projectglow.transformers.splitmultiallelics.VariantSplitter
 import io.projectglow.transformers.util.StringUtils
+
 import org.apache.spark.sql.DataFrame
-import io.projectglow.transformers.splitmultiallelics.VariantSplitter
+import org.apache.spark.sql.functions._
 
 /**
- * Implements DataFrameTransformer to transform the input DataFrame of varaints to an output
- * DataFrame of normalized variants (normalization is as defined in bcftools norm).
+ * Implements DataFrameTransformer to transform the input DataFrame of variants to an output
+ * DataFrame of normalized variants (normalization is as defined in vt normalize or bcftools norm).
+ *
+ * A path to reference genome containing .fasta, and .fai files must be provided
+ * through the reference_genome_path option.
+ *
+ * This transformer adds  a StructType normalizationStatus column to the DataFrame which contains
+ * the following fields:
+ *    - changed: A boolean fields whether the variant data was changed as a result of
+ *      normalization.
+ *    - errorMessage: An error message in case the attempt at normalizing the row
+ *       hit an error. In this case, the changed field will be set to false. If no errors occur
+ *       this field will be null.
+ *
+ * If the replace_columns option is false, the transformer also adds the StructType
+ * normalizationResult column to the DataFrame which contains the normalized start, end,
+ * referenceAllele, and alternateAlleles columns, whether changed or unchanged.  In case of
+ * error, normalizationResult will be null.
+ *
+ * If the replace_columns option is true (default), the transformer replaces the original start,
+ * end, referenceAllele, and alternateAlleles columns with the fields of normalizationResult (for
+ * no-error rows) and drops the normalizationResult column.
```

Comment:
Hm, it seems pretty dangerous to just drop all of the error rows. AFAIK, most other tools right out errors separately. The Spark way of doing that is to leave the errors so that people can filter out the errors / successes.

Reply:
No row will be dropped. The erroneous rows will not be touched and the normalizationStatus row is there to detect that they had errors.